modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -1978,13 +1980,109 @@ def sort_by(
+            if self.dtypes[columns[0]] == object:
+                # This means we are not sorting numbers, so we need our quantiles to not try
+                # arithmetic on the values.
+                method = "inverted_cdf"
+            else:
+                method = "linear"
+            from .utils import build_sort_functions
+
+            shuffling_functions = build_sort_functions(
+                self,
+                columns[0],
+                method,
+                ascending if not isinstance(ascending, list) else ascending[0],
+                **kwargs,
+            )
+            major_col_partition_index = self.columns.get_loc(columns[0])
+            cols_seen = 0
+            index = -1
+            for i, length in enumerate(self._column_widths_cache):
+                cols_seen += length
+                if major_col_partition_index <= cols_seen:
+                    index = i
+                    break
+            new_partitions = self._partition_mgr_cls.shuffle_partitions(
+                self._partitions,
+                index,
+                shuffling_functions.sample_function,
+                shuffling_functions.pivot_function,
+                shuffling_functions.split_function,
+                sort_function,
+            )
+            new_axes = self.axes
+            if kwargs.get("ignore_index", False):
+                new_axes[axis.value] = RangeIndex(len(new_axes[axis.value]))
+            else:
+                new_axes[axis.value] = self._compute_axis_labels_and_lengths(
+                    axis.value, new_partitions
+                )[0]
+            # Sometimes when performing a sort for `value_counts` the partitions have a MultiIndex
+            # but the _compute_axis_labels_and_lengths function gives us an Index of tuples instead
+            # of a MultiIndex.
+            if isinstance(self.axes[axis.value], pandas.MultiIndex) and isinstance(
+                new_axes[axis.value][0], tuple
+            ):
+                new_axes[axis.value] = pandas.MultiIndex.from_tuples(
+                    new_axes[axis.value].values
+                )
+            new_axes[axis.value] = new_axes[axis.value].set_names(
+                self.axes[axis.value].names
+            )
+            new_axes[axis.value ^ 1] = self.axes[axis.value ^ 1]
+            new_lengths = [None, None]
+            (
+                new_partitions,
+                new_lengths[axis.value],
+            ) = self._partition_mgr_cls.rebalance_partitions(new_partitions)
+            col_parts = new_partitions[0]
+            new_lengths[axis.value ^ 1] = [part.width() for part in col_parts]
+            new_modin_frame = self.__constructor__(
+                new_partitions, *new_axes, *new_lengths, self.dtypes
+            )
+            if kwargs.get("ignore_index", False):
+                new_modin_frame._propagate_index_objs(axis=0)
+            return new_modin_frame
+        elif axis == Axis.COL_WISE:
```

Comment:
I wonder if it should be an `else` instead of `elif`, cause in some super-weird case when `axis` is neither ROW nor COL it would blow up (I do agree that it shouldn't happen, but meh...)

Reply:
Sure - happy to do so!